### PR TITLE
Support RV32 configurations

### LINKED ISSUE #164: Allow RV32 BOOM Configs
- [x] Support RV32I
- [ ] Support RV32IF

### PR DESCRIPTION
**Pull request by jerryz123**
Resolves #164

### CODE REVIEW

File: src/main/scala/common/consts.scala [hunk 1]
```diff
@@ -23,7 +23,7 @@ import freechips.rocketchip.util.Str
 trait BOOMDebugConstants
 {
    val DEBUG_PRINTF        = false // use the Chisel printf functionality
-   val COMMIT_LOG_PRINTF   = false // dump commit state, for comparision against ISA sim
+   val COMMIT_LOG_PRINTF   = true // dump commit state, for comparision against ISA sim
```
Comment (abejgonzalez) [inline src/main/scala/common/consts.scala]: Keep disabled for merge. Forgot to undo?

File: src/main/scala/common/parameters.scala [hunk 2]
```diff
@@ -127,6 +126,8 @@ trait HasBoomCoreParameters extends freechips.rocketchip.tile.HasCoreParameters
    val usingFDivSqrt = boomParams.fpu.isDefined && boomParams.fpu.get.divSqrt
 
    val mulDivParams = boomParams.mulDiv.getOrElse(MulDivParams())
+   // TODO: Allow this
```
Comment (abejgonzalez) [inline src/main/scala/common/parameters.scala]: Extraneous comment since it now "works"

File: src/main/scala/system/Configs.scala [hunk 3]
```diff
@@ -27,8 +27,11 @@ class BoomConfig extends Config(
    new DefaultBoomConfig ++
    new WithNBoomCores(1) ++
    new WithoutTLMonitors ++
-   new freechips.rocketchip.system.BaseConfig)
+      new freechips.rocketchip.system.BaseConfig)
```
Comment (abejgonzalez) [inline src/main/scala/system/Configs.scala]: Spacing?

File: src/main/scala/system/Configs.scala [hunk 4]
```diff
@@ -119,6 +124,17 @@ class TracedSmallBoomConfig extends Config(
    new WithoutTLMonitors ++
    new freechips.rocketchip.system.BaseConfig)
 
+//RV32IMAC TODO: Support FP
```
Comment (abejgonzalez) [inline src/main/scala/system/Configs.scala:127]: I feel like this should be mentioned in the future work section of the docs or at the very least in a GH issue

**Comment by abejgonzalez** [comment]
Lgtm other than the failing csmith tests which I am looking into. Ill approve once that is figured out.

**Review by abejgonzalez — APPROVED** [review]
Lgtm. CI was failing since the emulator was taking ~16m on that seed. So I increased the timeout and reduced the number of runs to reduce the overall runtime. I think that expanding the timeout further would be unwarranted (just in case we hit a "looping" case where the sim is not in a pipeline hang but is making no progress)

Just change the recommendations then I think this is mergable

File: src/main/scala/system/Configs.scala (hunk 4 above)
Comment (jerryz123) [reply src/main/scala/system/Configs.scala:127]: I'll add to issue

**Pull request merged**